### DISCUSSION TITLE
Why does the handler remove the body from the response?

### DISCUSSION THREAD

**Discussion by evgenoir** [mswjs/msw#1255] [Q&A] [answered]
I was trying to test some handler using **_fetch()_** and noticed that when I set the body in the handler with **_ctx.json()_**, **_ctx.body()_** or directly with _**res.body = "smth"**_, I didn't get any body in response and '_bodyUsed_' field of the received response was always '_false_'. I thought _**fetch()**_ was the cause (browser vs Node) and tested _**fetch()**_ with _**setupWorker()**_ and _**setupServer()**_ with the same result.
I've created an [repository outlining this problem](https://github.com/evgenoir/msw_fetch)!
So, my question is: Where does the body disappear to and how to preserve it?

**Comment by kettanaito** [comment, marked as answer]
Hi, @evgenoir. 

Your handler seems to be working correctly. The payload you provide to the `ctx.json` will become the response body _once it's read_. You output the `response` itself into the console, and it will indeed have no body because you haven't read it yet. 

Response body is kept in what is called `ReadableStream`, and you don't know the contents of that stream until you explicitly read them. That's how you do it with fetch:

```js
// src/App.js
const response = await fetch('/test', {
    method: 'POST',
    headers: new Headers({
        authorization: "Bearer some_token #1"
    })
})

// Read the response body.
// The body itself is mocked via MSW while it's active.
const text = await response.text()

console.log(response.bodyUsed) // true
console.log(text) // "test OK"
```

Since you're returning a plain text body (using `ctx.body` without an explicit `Content-Type` header), you should be reading your responses as text. That's what `response.text()` does. Response body reading happens on the client (your own code, not mocks).

> I highly recommend reading through the [fetch API Response](https://developer.mozilla.org/en-US/docs/Web/API/Response) instance on MDN.

**Reply by evgenoir** [reply]
Oh, stupid mistake. One could guess. Thanks for the help!
